include/counter.h

```diff
@@ -132,7 +205,7 @@ static inline int counter_set_alarm(struct device *dev,
  *
  * @param dev Pointer to the device structure for the driver instance.
  *
- * @retval 1 if the counter interrupt is pending.
+ * @retval 1 if the counter any interrupt is pending.
  * @retval 0 if no counter interrupt is pending.
  */
 __syscall int counter_get_pending_int(struct device *dev);
```

Comment:
As we are at reworking the whole API, can we challenge that function? is it really useful?

Reply:
I am fairly new to zephyr. I've seen that it is mainly used to determine wakeup source. Is it only for testing?

Reply:
I am not sure why it's there. We have it also for gpio, aio... But it's unused anywhere, I don't see why user code would need to know about pending interrupts, moreover that we are fully interrupt based. Unless you lock interrupt for a long time, and want to behave as a polling-based code. 
The only code using it is a PM test. I kind of get why it does that there, but perhaps there is another way.
Let's see next device driver API meeting, will put that on the agenda.

Reply:
What's the conclusion, is this function to stay?